### PR TITLE
Add pipeline for validating and publishing the wasmcloud helm chart

### PR DESCRIPTION
**Pull request by joonas**
## Feature or Problem

Adds a pipeline that validates the helm chart (added in #788) and (on `chart-vX.X.X` push) publishes the helm chart to `ghcr.io/wasmcloud/wasmcloud-chart`.

## Related Issues

#684

## Testing

Built on platform(s)
- [ ] x86_64-linux
- [ ] aarch64-linux
- [ ] x86_64-darwin
- [ ] aarch64-darwin
- [ ] x86_64-windows

Tested on platform(s)
- [ ] x86_64-linux
- [ ] aarch64-linux
- [ ] x86_64-darwin
- [ ] aarch64-darwin
- [ ] x86_64-windows

### CODE REVIEW

File: .github/workflows/wasmcloud-chart.yml
Comment (connorsmith256) [inline .github/workflows/wasmcloud-chart.yml:51]: Nit: isn't checking the event name redundant? I'm not sure there's a way for the ref to be a tag without the event being a push

File: .github/workflows/wasmcloud-chart.yml
Comment (connorsmith256) [inline .github/workflows/wasmcloud-chart.yml:82]: I don't have any experience publishing charts, but did notice for the [old chart](https://github.com/wasmCloud/wasmcloud-otp/blob/main/.github/workflows/release.yml) we use the `helm/chart-releaser-action` action. Is there a reason we shouldn't use that anymore?

**Review by connorsmith256 — COMMENTED** [review]
LGTM, with the caveat that I don't have domain knowledge on helm charts. Can approve if needed, but will defer to others for now

File: .github/workflows/wasmcloud-chart.yml
Comment (joonas) [reply .github/workflows/wasmcloud-chart.yml:51]: > Nit: isn't checking the event name redundant? I'm not sure there's a way for the ref to be a tag without the event being a push

It is, I'll drop the `github.event_name` check here.

File: .github/workflows/wasmcloud-chart.yml
Comment (joonas) [reply .github/workflows/wasmcloud-chart.yml:82]: So the [`helm/chart-releaser-action`](https://github.com/helm/chart-releaser-action) is primarily intended for publishing charts to be self-hosted on GitHub Pages, which is not how we intend to host the chart. Instead, we're looking to publish the chart onto GitHub Container Registry as an OCI artifact.

The `chart-releaser-action` has an [issue open regarding adding support for OCI registries](https://github.com/helm/chart-releaser-action/issues/107), which would let us use it in the future should they choose to implement that functionality.

Comment (connorsmith256) [reply .github/workflows/wasmcloud-chart.yml:82]: Ah, got it! Thanks 👍

**Review by thomastaylor312 — APPROVED** [review]

**Pull request merged**
